### PR TITLE
add map versioning checks

### LINKED ISSUE #181: Implement a map version and check against it when loading a map

### PR DESCRIPTION
**Pull request by tlein**
Resolves #181

### CODE REVIEW

File: src/duckling/version.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    major : string,
+    minor : string
+};
+
+/**
+ * Get the major version for the given map version.
+ * @param mapVersion Map version to check, MAP_VERSION will be used if not supplied
+ * @return The major version for the given map version
+ */
+export function majorMapVersion(mapVersion? : string) : string {
```
Comment (jeffswenson) [inline src/duckling/version.ts]: These should return integers since you throw errors if they aren't integers.

File: src/duckling/version.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+export const VERSION = "0.0.1";
+export const MAP_VERSION = "1.1";
```
Comment (jeffswenson) [inline src/duckling/version.ts]: I would implement a completely different API. You should write a compareVersions(actualVersion, expectedVersion). It should return a Compatibile, MinorIncompatible, MajorIncompatible enum.

Comment (tlein) [reply src/duckling/version.ts]: fixed

File: src/duckling/version.ts (hunk 1 above)
Comment (tlein) [reply src/duckling/version.ts]: fixed

**Pull request merged**